move all logic to the cli factory side

joaomcteixeira opened the pull request:
@PicoCentauri 
What do you think of this implementation?

After our discussion on the email regarding the fact that MDA Analysis classes are not loaded unless strictly requested with an `import` statement, I thought that there is actually no benefit in the `@add_to_CLIs` decorator strategy. Therefore, we could just use the `add_to_CLIs` as a function inside the `cli_main` factory and `inspect` the modules manually based on the `__all__` list on the `MDAnalysis.analysis.__init__`.

In the first commit of this PR, I am importing `add_to_CLIs` and `analysis_interfaces` from MDA as they are in your fork. But if you agree with this strategy, I think it is best to just move everything to the `cli` side. Also, in this way, the `cli` logic gets completely separated from the MDA and it does not overload the main package for those users that don't want to use the client.

If you run `cli_main.py` it works :wink:

What are your thoughts?

PicoCentauri commented on cli_main.py:
```diff
@@ -9,12 +9,24 @@
 import MDAnalysis as mda
-from MDAnalysis.analysis.base import AnalysisBase, add_to_CLIs
+from MDAnalysis.analysis import __all__
 from MDAnalysis.analysis import analysis_interfaces
+from MDAnalysis.analysis.base import add_to_CLIs, AnalysisBase
```
Move the logic from the fork to this repo.

PicoCentauri commented:
It is really good idea being completely independent from the main repository 💪. Would be nice to extend this PR and include the logic here. 

I just made a little dummy comment to keep track. But I know that you know what you are doing.

joaomcteixeira commented:
@PicoCentauri done.

The whole `CLI` creation logic is now here. I directly took it from our previous implementation that was stored in the [CLI branch](https://github.com/PicoCentauri/mdanalysis/tree/CLI/package/MDAnalysis/analysis), you as co-author in this commit.

I have cleaned slightly the file and added very little documentation. I have not altered anyway the logic of the functions. We will implement the other discussions on that regard in another PR. 

There are a couple of issues more I identified with this PR, but I will do so in another issue separately as they relate also to upstream implementations.

@PicoCentauri , if you agree, please merge this PR. :wink:
